frontends/web/src/routes/exchange/components/buysell.tsx

```diff
@@ -0,0 +1,155 @@
+import { useNavigate } from 'react-router-dom';
+import { useLoad } from '@/hooks/api';
+import * as exchangesAPI from '@/api/exchanges';
+import { useContext, useEffect, useState } from 'react';
+import { ExchangeSelectionRadio } from './exchangeselectionradio';
+import { Button } from '@/components/forms/button';
+import { useTranslation } from 'react-i18next';
+import { getVersion } from '@/api/bitbox02';
+import style from '../exchange.module.css';
+import { AppContext } from '@/contexts/AppContext';
+import { TInfoContentProps } from './infocontent';
+
+type TProps = {
+  accountCode: string;
+  deviceIDs: string[];
+  selectedRegion: string;
+  onSelectExchange: (exchange: string) => void;
+  selectedExchange: string;
+  goToExchange: () => void;
+  showBackButton: boolean;
+  action: exchangesAPI.TExchangeAction
+  setInfo: (into: TInfoContentProps) => void;
+}
+
+export const BuySell = ({
+  accountCode,
+  deviceIDs,
+  selectedRegion,
+  onSelectExchange,
+  selectedExchange,
+  goToExchange,
+  showBackButton,
+  action,
+  setInfo,
+}: TProps) => {
+  const { t } = useTranslation();
+  const { setFirmwareUpdateDialogOpen } = useContext(AppContext);
+
+  const exchangeDeals = useLoad(() => exchangesAPI.getExchangeDeals(action, accountCode, selectedRegion), [action, selectedRegion]);
+  const [exchangeDealsError, setExchangeDealsError] = useState('');
+  const [updateFirmwareToSell, setUpdateFirmwareToSell] = useState(false);
+  // FIXME: we should make a BE `can-sell` endpoint
+  const versionInfo = useLoad(() => getVersion(deviceIDs[0]), [deviceIDs[0]]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (exchangeDeals?.success === false) {
```

Comment:
I noticed this small render bug / hiccup: 

https://github.com/user-attachments/assets/f10f593e-89cf-4ad3-87f7-fb0d7ea2a21f

I believe this was caused by the `exhcangeDealsError` state that was set when we click on "Sell" (while it has no sell option). That state wasnt reset on changing the tab (clicking "Buy"). So this error message is tehcnically for "Sell" (? 🤔thus actually an incorrect message).

Suggestion: 
-reset exchangeDealsError state on tab change 
-& optionally add a loading state instead of showing the (inaccurate) error message. 

WDYT?

Reply:
Nice catch, thanks! I added a loading `Skeleton` and updated the `useLoad` hook to set back the response to `undefined` while waiting for a new result. Hope that is ok.